xmltest.cpp

```diff
@@ -431,7 +435,8 @@ int main( int argc, const char ** argv )
 		XMLTest( "Programmatic DOM", 2, doc->FirstChildElement()->LastChildElement( "sub" )->IntAttribute( "attrib" ) );
 		XMLTest( "Programmatic DOM", "& Text!",
 				 doc->FirstChildElement()->LastChildElement( "sub" )->FirstChild()->ToText()->Value() );
-		XMLTest("User data", (void*)2 == comment->GetUserData(), true, false);
+		XMLTest("User data", true, &dummyValue == comment->GetUserData(), false);
+		XMLTest("User data", dummyInitialValue, dummyValue, false);

```

Comment:
@leethomason It tests that the value pointed to hasn't changed. It's how "opaque" values should work - you pass its address anywhere but the value itself doesn't change because that "anywhere" has no idea of how to change the value.